platforms/android/src/org/crosswalk/engine/XWalkCordovaUiClient.java

```diff
@@ -172,12 +175,15 @@ public void onPageLoadStopped(XWalkView view, String url, LoadStatus status) {
     // File Chooser
     @Override
     public void openFileChooser(XWalkView view, final ValueCallback<Uri> uploadFile, String acceptType, String capture) {
-        uploadFile.onReceiveValue(null);
+        if (mFileChooser == null) {
+            mFileChooser = new XWalkFileChooser(parentEngine.cordova.getActivity());
```

Comment:
About the code here, the problem is not the different between the application context and the activity. The point is that the context inside of XWalkView may not be the activity that the developer wants to launch the file chooser.